verification/curator-service/ui/src/components/SourceTable.tsx

```diff
@@ -77,14 +94,40 @@ export default class SourceTable extends React.Component<{}, SourceTableState> {
         })
     }
 
+    createSourceFromRowData(rowData: TableRow) {
+        return {
+            _id: rowData._id,
+            name: rowData.name,
+            origin: {
+                url: rowData.url
+            }
+        }
+    }
+
+    validateRequired(field: String) {
+        return field?.trim() !== "";
+    }
+
     render() {
         return (
             <Paper>
                 <MaterialTable
                     columns={[
                         { title: 'ID', field: '_id', editable: "never" },
                         { title: 'Name', field: 'name' },
-                        { title: 'URL', field: 'url' },
+                        {
+                            title: 'URL', field: 'url',
+                            editComponent: (props) =>
+                                (<TextField
+                                    type="text"
+                                    size="small"
+                                    fullWidth
+                                    placeholder="Source URL"
```

Comment:
nit: just "URL" to match the column title

Reply:
SG, updated.